zookeeper/source/zk_nn_client.cc

```diff
@@ -1421,27 +1442,73 @@ bool ZkNnClient::add_block(const std::string &file_path,
   return true;
 }
 
-bool ZkNnClient::add_block_group(const std::string &fileName,
+bool ZkNnClient::add_block_group(const std::string &filePath,
                      u_int64_t &block_group_id,
                      std::vector<std::string> &dataNodes,
-                     std::vector<u_int64_t> &storageBlockIDs,
+                     std::vector<char> &blockIndices,
                      uint32_t total_num_storage_blocks) {
-    FileZNode znode_data;
-    read_file_znode(znode_data, fileName);
+  FileZNode znode_data;
+  read_file_znode(znode_data, filePath);
+  std::vector<u_int64_t> storage_block_ids;
+  // Generate the block group id and storage block ids.
+  block_group_id = generate_block_group_id();
+  for (u_int64_t i = 0; i < total_num_storage_blocks; i++) {
+    storage_block_ids.push_back(generate_storage_block_id(
+        block_group_id, i));
+  }
+
+  // Log them for debugging purposes.
+  LOG(INFO) << "Generated block group id " << block_group_id << "\n";
+  for (auto storageBlockID : blockIndices)
```

Comment:
Why are we getting storageBlockIDs from a blockIndices array if we have a storage block_ids array.

Reply:
That's a mistake made by CLion auto refactoring. it should be ```storage_block_ids``` instead of ```blockIndices```